// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.dash_backend";
option java_outer_classname = "DashBackendProto";
option objc_class_prefix = "DBE";

package dash_backend;

// The greeting service definition.
service Dash {
  // Sends a greeting
  rpc Dash_Get(Key) returns (Value){}
  rpc Dash_Put(File) returns (PutReply){}
}

// The request message containing the user's name.
message Key {
  string key = 1;
}

message Value {
  bytes value = 1;
}

message File {
  string key = 1;
  bytes value = 2;
}

message PutReply {
    string reply_message = 1;
}
